File: Selenium/Activities/Activity16.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select

# Start the Driver
with webdriver.Firefox() as driver:
    # Navigate to the URL
    driver.get("https://training-support.net/webelements/selects")
    # Print the title of the page
    print("Page title is: ", driver.title)

    # Find the dropdown
    dropdown = driver.find_element(By.CSS_SELECTOR, "select.h-10")
    # Pass the WebElement to the Select object
    singleSelect = Select(dropdown)

    # Select the second option using visible text
    singleSelect.select_by_visible_text("Two")
    # Print the selected option
    print("Second option: " + singleSelect.first_selected_option.text)

    # Select the third option using index
    singleSelect.select_by_index(3)
    # Print the selected option
    print("Third option: " + singleSelect.first_selected_option.text)

    # Select the fourth option using value attribute
    singleSelect.select_by_value("four")
    # Print the selected option
    print("Fourth option: " + singleSelect.first_selected_option.text)

    # Print all the options
    allOptions = singleSelect.options
    print("Options in the dropdown: ")
    for option in allOptions: 
        print(option.text)

File: Selenium/Activities/Activity16.java
package activities;

import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity16 {
    public static void main(String[] args) {
        // Create a new instance of the Chrome driver
        WebDriver driver = new ChromeDriver();

        // Open the page
        driver.get("https://training-support.net/webelements/selects");
        // Print the title of the page
        System.out.println("Page title is: " + driver.getTitle());

        // Find the dropdown
        WebElement dropdown = driver.findElement(By.cssSelector("select.h-10"));
        // Pass the WebElement to the Select object
        Select singleSelect = new Select(dropdown);

        // Select the second option using visible text
        singleSelect.selectByVisibleText("Two");
        // Print the selected option
        System.out.println("Second option: " + singleSelect.getFirstSelectedOption().getText());

        // Select the third option using index
        singleSelect.selectByIndex(3);
        // Print the selected option
        System.out.println("Third option: " + singleSelect.getFirstSelectedOption().getText());

        // Select the fourth option using value attribute
        singleSelect.selectByValue("four");
        // Print the selected option
        System.out.println("Fourth option: " + singleSelect.getFirstSelectedOption().getText());

        // Print all the options
        List<WebElement> allOptions = singleSelect.getOptions();
        System.out.println("Options in the dropdown: ");
        for(WebElement option : allOptions) {
            System.out.println(option.getText());
        }

        // Close the browser
        driver.quit();
    }
}
